src/v2/keychains.js

```diff
@@ -82,9 +88,12 @@ Keychains.prototype.updatePassword = function(params, callback) {
           continue;
         }
         try {
-          const decryptedPrv = this.bitgo.decrypt({ input: oldEncryptedPrv, password: params.oldPassword });
-          const newEncryptedPrv = this.bitgo.encrypt({ input: decryptedPrv, password: params.newPassword });
-          changedKeys[key.pub] = newEncryptedPrv;
+          const updatedKeychain = this.changeSingleKeychainPassword({
+            keychain: key,
+            oldPassword: params.oldPassword,
+            newPassword: params.newPassword
+          });
+          changedKeys[updatedKeychain.pub] = updatedKeychain.encryptedPrv;
         } catch (e) {
           // catching an error here means that the password was incorrect and hence there is nothing to change
```

Comment:
Do we want to raise here to inform the caller that the (old?) password passed to the fcn was incorrect and that no changes were applied?

Reply:
Can you catch specific errors in javascript or do we need to resort to generically catching _any_ error that the code above could throw?

Reply:
> Do we want to raise here to inform the caller that the (old?) password passed to the fcn was incorrect and that no changes were applied?

No, since this needs to loop through all keychains, and only change the ones which happened to have the same wallet passphrase as the user login password. If the user login password doesn't match the wallet passphrase, it should be skipped, and no error should be thrown. The frontend relies on this behavior, so it cannot be changed without a major version bump.

> Can you catch specific errors in javascript or do we need to resort to generically catching _any_ error that the code above could throw?

Yes, you can by using `instanceof`. But we are not using custom error types (yet) in bitgojs, so we would need to catch all and check the error message, which gets ugly

Reply:
I'm worried that if we throw an error here it will break other users who are relying on this not throwing an error.

Reply:
> I'm worried that if we throw an error here it will break other users who are relying on this not throwing an error.

Agreed, we cannot change the behavior without a major version bump as the front end relies on a password mismatch causing the keychain to be skipped instead of throwing an error.